prism/src/main/java/org/apache/falcon/resource/proxy/SchedulableEntityManagerProxy.java

```diff
@@ -378,6 +393,34 @@ protected APIResult doExecute(String colo) throws FalconException {
         return consolidateResult(results, APIResult.class);
     }
 
+    private void isEntityPartOfAnExtension(Entity entity) {
+        String tags = entity.getTags();
+        checkExtensionJobExist(tags);
+    }
+
+
+    private void doesEntityHasExtensionJobTag(Entity entity) {
+        String tags = entity.getTags();
+        if (StringUtils.isNotBlank(tags)) {
+            String jobName = ExtensionManager.getJobNameFromTag(tags);
+            if (StringUtils.isNotBlank(jobName)) {
+                throw FalconWebException.newAPIException("Entity submit is not allowed on an entity having extension "
```

Comment:
The error message can also say, that such entities need to be submitted as extension jobs.